File: Source/Fayde.Server/Media/Animation/BeginStoryboard.cs
using Fayde.Core;
using Fayde.Xaml.Metadata;

namespace Fayde.Media.Animation
{
    public class BeginStoryboard : TriggerAction
    {
        public static readonly PropertyDescription StoryboardProperty = PropertyDescription.Register("Storyboard", typeof(Storyboard), typeof(BeginStoryboard), true);
    }
}

File: Source/Fayde.Client/Media/Animation/BeginStoryboard.ts
/// <reference path="../../Core/Triggers.ts" />

module Fayde.Media.Animation {
    export class BeginStoryboard extends TriggerAction {
        static StoryboardProperty: DependencyProperty = DependencyProperty.Register("Storyboard", () => Animation.Storyboard, BeginStoryboard);
        Storyboard: Animation.Storyboard;

        Fire() {
            var sb = this.Storyboard;
            if (sb) sb.Begin();
        }
    }
    Fayde.RegisterType(BeginStoryboard, "Fayde.Media.Animation", Fayde.XMLNS);
    Xaml.Content(BeginStoryboard, BeginStoryboard.StoryboardProperty);
}